src/etl/normalizer.py

```diff
@@ -0,0 +1,223 @@
+import re
+import polars as pl
+import pycountry
+from logging import Logger
+
+from utilities.units_converter import converter_map
+
+
+class Normalizer:
+    def __init__(self, config: dict, logger: Logger) -> None:
+        self.logger = logger
+        self.data_config = config["data_config"]
+        self.data_columns = config["data_columns"]
+        self.data_normalizer = config["data_normalizer"]
+
+    def execute(self, df_base: pl.DataFrame) -> pl.DataFrame:
+        """
+        Execute all the methods defined in the normalizer.
+        """
+        pipeline = [
+            self.normalize_inches_columns,
+            self.normalize_pound_columns,
+            self.convert_units_to_float,
+            self.convert_currency,
+            self.filter_wrong_upcs,
+            self.convert_upc_to_ean13,
+            self.rename_columns,
+            self.normalize_country_column,
+            self.format_date_columns,
+            self.convert_int_columns,
+        ]
+
+        for step in pipeline:
+            df_base = step(df_base)
+
+        return df_base
+
+    def convert_units(
+        self,
+        df_base: pl.DataFrame,
+        valid_units: str,
+        pattern: re.Pattern,
```

Comment:
If pattern is only supposed to be a compiled string-based regex, use `re.Pattern[str]` as your hint.

https://docs.python.org/3/library/re.html#re.Pattern